services/logging.py

```diff
@@ -80,6 +80,13 @@ class MissingVariableErrorFilter(logging.Filter):
         # are missing when pages are requested by a `RequestFactory` instance.
         "csp_nonce",  # csp.middleware.CSPMiddleware
         "template_name",  # jobserver.middleware.TemplateNameMiddleware
+        # And some template variables are just a pain, because of how we use them.
+        "name",  # jobserver.context_processors.nav
+        "q",
+        "get_staff_url",
```

Comment:
As far as I can see, `get_staff_url` is currently only producing a warning from one test:
```
FAILED tests/integration/test_interactive.py::test_interactive_publishing_report_success - UserWarning: Exception while resolving variable 'get_staff_url' in template 'staff/report/detail.html'.
```

If that's correct, then it would be worth just editing that template (I think it should be a simple fix).